### PR TITLE
Make it so that the max_function_length style checker can ignore func…

### PR DESCRIPTION
**Pull request by sargun**
…tions

### CODE REVIEW

**Comment by gadgetci** [comment]
**xref** failed with exit status: 1

File: src/elvis_style.erl [hunk 1]
```diff
@@ -517,22 +517,25 @@ max_module_length(Config, Target, RuleConfig) ->
             []
     end.
 
+-type max_function_length_config() :: #{ignore_functions => [function_spec()]
+                                        max_length => non_neg_integer()}.
```
Comment (gadgetci) [inline src/elvis_style.erl]: According to **Xref**:
> syntax error before: max_length

File: src/elvis_style.erl [hunk 2]
```diff
@@ -517,22 +517,25 @@ max_module_length(Config, Target, RuleConfig) ->
...
 -spec max_function_length(elvis_config:config(),
                           elvis_file:file(),
-                          empty_rule_config()) ->
+                          max_function_length_config()) ->
```
Comment (gadgetci) [inline src/elvis_style.erl]: According to **Xref**:
> type max_function_length_config() undefined

File: src/elvis_style.erl (hunk 1 above)
Comment (elbrujohalcon) [inline src/elvis_style.erl]: You're missing a comma here

File: src/elvis_style.erl [hunk 3]
```diff
@@ -517,22 +517,25 @@ max_module_length(Config, Target, RuleConfig) ->
             []
     end.
 
+-type max_function_length_config() :: #{ignore_functions => [function_spec()]
```
Comment (elbrujohalcon) [inline src/elvis_style.erl]: To preserve module style, you should define this type above, close to the other types defined in the module

**Comment by sargun** [comment]
Any idea why common test is throwing the following:
```
3c075477e55e:elvis_core sdhillon$ rebar3 ct
===> Verifying dependencies...
===> Compiling elvis
===> Failed to restore /Users/sdhillon/repos/lashup/_checkouts/elvis_core/_build/test/lib/elvis/.rebar3/erlcinfo file. Discarding it.

===> Compiling _build/test/lib/elvis/test/examples/fail_macro_module_names.erl failed
_build/test/lib/elvis/test/examples/fail_macro_module_names.erl:15: redefining predefined macro 'FUNCTION_NAME'

_build/test/lib/elvis/test/examples/fail_macro_module_names.erl:21: Warning: missing specification for function module_name/0
_build/test/lib/elvis/test/examples/fail_macro_module_names.erl:25: Warning: missing specification for function function_name/0
_build/test/lib/elvis/test/examples/fail_macro_module_names.erl:30: Warning: missing specification for function build_binary/0
_build/test/lib/elvis/test/examples/fail_macro_module_names.erl:38: Warning: missing specification for function no_errors/0

```

**Comment by sargun** [comment]
Fixed.

**Pull request merged**